### PR TITLE
AMP-82205 de-emphasised environments based ampli initialization, and …

### PR DESCRIPTION
**Pull request by akramarev**
…recommended to use client.apiKey instead. Normalized AMPLITUDE_API_KEY usage across the docs.

## Deadline

Not urgent.

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [X] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

**Review by rctbusk — APPROVED** [review]

**Review by markzegarelli — APPROVED** [review]

File: docs/data/ampli/migration.md [hunk 1]
```diff
@@ -178,12 +178,14 @@ amplitude.init(AMPLITUDE_API_KEY);
 ```typescript
 import { ampli, SongPlayed } from './ampli';
 
-// Initialize with an Environment from Data
+// Initialize with Amplitude API Key
 ampli.load({
-  environment: 'my-production'
```
Comment (ondrej) [inline docs/data/ampli/migration.md:183]: Is there a reason we're leaving `environment` in the some snippets while removing it in others?

File: docs/data/sdks/android/ampli.md [hunk 2]
```diff
@@ -113,19 +113,19 @@ Initialize Ampli in your code. The `load()` method accepts configuration option
     ```java
     import com.amplitude.ampli.*;
 
-    Ampli.getInstance().load(appContext, new LoadOptions()
-      .setEnvironment(Ampli.Environment.PRODUCTION)
+    Ampli.getInstance().load(this, new LoadOptions()
+      .setClient(new LoadClientOptions().setApiKey(AMPLITUDE_API_KEY))
```
Comment (ondrej) [inline docs/data/sdks/android/ampli.md:117]: The syntax for setting API key in Browser isn't terrible (though still lots of nested objects, etc.) but gets a bit gnarlier in other languages. Wasn't too bad when it was the 1% case but now that it's the 99% case needing all this boilerplate to set the API key is a bummer (as opposed to it being a top-level config option like environment used to be)

Comment (akramarev) [reply docs/data/sdks/android/ampli.md:117]: Good point, I agree that the ampli load in Java specifically looks a bit wordy now, though we have plans to improve the interface near term across the board.

File: docs/data/ampli/migration.md (hunk 1 above)
Comment (akramarev) [reply docs/data/ampli/migration.md:183]: re-checked together, seems everything is updated now
